### ISSUE TITLE
Обновление от 29.07.2019

### ISSUE THREAD

**Issue by SBProX** [SBProX/SBProX-Dev#446]
1. Исправлено изменение цены линии при открытии ее свойств: https://github.com/SBProX/SBProX-Dev/issues/440
2. Исправлено отображение объема на кластерном графике при низком масштабе: https://github.com/SBProX/SBProX-Dev/issues/416
3. Исправлено затемнение шрифта при включении режима плотности объемов: https://github.com/SBProX/SBProX-Dev/issues/443
4. Исправлен визуальный баг, при котором было несовпадение ячеек Cluster profile и индикатора Period volume profile: https://github.com/SBProX/SBProX-Dev/issues/325
5. В платформе доработана система масштабирования (пока что в **ЭКСПЕРИМЕНТАЛЬНОМ** режиме):
- масштабирование стало более плавным
- сильно уменьшились пороги сжатия графика по обоим осям (Внимание! При сильном сжатии графика на большом периоде данным возможны подтормаживания!)
- при детальном рассмотрении графика могут быть сдвиги расстояния между элементами, но не значительные.

Данный пункт был реализован после ознакомления с тикетами:
https://github.com/SBProX/SBProX-Dev/issues/399
https://github.com/SBProX/SBProX-Dev/issues/319
Так же данная система мастабирования, если понравится пользователям, будет продублирована в SBProV2 через некоторое время.

**Comment by test-777** [comment]
Спасибо за масштабирование, те рывки которые докучали в прошлой версии теперь в прошлом. 
Возможность сжатия графика тоже весьма порадовала - на монитор уменьшается больший диапазон времени что весьма полезно!

**Comment by SBProX** [comment]
@test-777 Спасибо за позитивный отзыв, хоть кто-то написал.

**Comment by Flawmaster** [comment]
Спасибо за масштабирование! Сейчас реализовано идеально. Намного приятнее стало работать, т.к. график не сжимается рывками при малейшем движении мышкой.

**Comment by SBProX** [comment]
@Flawmaster Рады стараться. Спасибо.

**Comment by vasiliyyun** [comment]
Коллеги, не совсем разделяю радости по поводу нововведения с масштабированием. Я торгую внутри дня, использую мелкий тайм. в процессе мне важна скорость. Теперь чтобы масштабировать график нужно больше времени (за счет плавности), нужны лишние движения. По поводу самого масштабирования. Вернулась старая проблема. После масштабирования слетает выставленная настройка по масштабу по вертикали. Т.е. после каждого увеличения/уменьшения масштаба, приходится снова выставлять оптимальный масштаб по вертикали, очень неудобно!!!

**Comment by vasiliyyun** [comment]
Мне также важно точное значение максимума/минимума. В программе нет функции приманичивания линий к эсктремумам (как это реализовано в  некоторых других платформах). Эту проблему я решал с помощью масштабирования графика, где при сильном масштабе вручную можно выставить линию с точностью до тика. Теперь чтобы сделать это мне нужно тратить больше времени. Раньше нужного значения масштаба можно было достичь прокрутив колесико в полный оборот три-четыре раза, теперь его нужно крутить и крутить. В общем это очень неудобно! Плафторма ведь должна быть оптимальной для всех пользователей. Мне вот например не важно сколько графика у меня помещается в монитор. И для меня эта функция абсолютно бесполезна, кроме того, она мне теперь мешает работать в привичном режиме.

**Comment by vasiliyyun** [comment]
Да, и еще добавлю про вертикальное масштабирование. После нескольких манипуляцию с уменьшением/увеличением графика, он превращается в "линию", на столько сбивается вертикальный масштаб. Вы ведь раньше устраняли эту проблему, потому как многие на это жаловались. Теперь она вернулась.

**Comment by Flawmaster** [comment]
> В программе нет функции приманичивания линий к эсктремумам

Если только эту функцию сделают опциональной, т.к. не всем это удобно, потому что у инструментов рисования и так ход ровно по тикам.

**Comment by vasiliyyun** [comment]
В общем потестировав нововведенную систему масштабирования  можно сделать следующие выводы.
Про то, что увеличение/уменьшение графика стало осуществляться медленнее - факт. Непривычно, отсюда и не удобно. Может сделать эту плавность опциональной? Мне, например, прежний вариант больше нравился.
Про масштабирование по вертикали. Оценил фишку. Это конечно классно, что теперь можно уместить большой период растянутый по вертикали. Но! Масштаб по вертикали все же сбивается, и это не удобно. Что если так В текущий момент, если максимально уменьшить график, то масштаб по вертикали тоже уменьшается (как бы сплющивается). Можно ли сделать так, чтобы при увеличении графика (после уменьшения), масштаб по вертикали также расширялся бы. И при возврате графика в нормальное положение, вертикальный масштаб возвращался бы изначально настроенное положение. Так было бы гораздо удобнее.

**Comment by SBProX** [comment]
@vasiliyyun касательно непропорционального масштабирования колесом мыши - это мы поправим. Насчет остального - всем не угодишь. **Большое кол-во пользователей** давно просили сделать масштабирование более плавным - мы сделали. Мы не можем туда-сюда переделывать под конкретные пожелания.
